src/app/weak-supervision/components/weak-supervision/weak-supervision.component.ts

```diff
@@ -490,4 +492,11 @@ export class WeakSupervisionComponent implements OnInit, OnDestroy {
     })
 
   }
+
+  navigateToSettings() {
+    this.router.navigate(["../settings"], {
+      relativeTo: this.activatedRoute,
+    });
+    localStorage.setItem("openModal", "true");
```

Comment:
i think i would switch these so the localstorage is set before the route is changed to prevent some race condition